src/persist/src/runtime.rs

```diff
@@ -125,97 +124,162 @@ where
             }
         }
     });
-
-    // Construct the client.
-    let handles = Mutex::new(Some(RuntimeHandles {
+    let handles = RuntimeHandle(RuntimeHandleInner::Full {
+        id,
         impl_handle,
         ticker_handle,
         ticker_pool: pool.clone(),
-    }));
-    let core = RuntimeCore {
-        handles,
-        tx,
-        metrics,
+    });
+
+    // Explictly drop the pool guard so the lifetime is obvious.
+    drop(pool_guard);
+
+    Ok(RuntimeClient::new(handles, tx, metrics))
+}
+
+/// Starts a read-only runtime in a [std::thread].
+///
+/// This returns a clone-able client handle. The runtime is stopped when any
+/// client calls [RuntimeClient::stop] or when all clients have been dropped.
+///
+/// If Some, the given [tokio::runtime::Runtime] is used for IO and cpu heavy
+/// operations. If None, a new Runtime is constructed for this. The latter
+/// requires that we are not in the context of an existing Runtime, so if this
+/// is the case, the caller must use the Some form.
+//
+// TODO: The rust doc above is still a bit of a lie. Actually use this runtime
+// for all IO and cpu heavy operations.
+//
+// TODO: The whole story around Runtime usage in persist is pretty awkward and
+// still pretty unprincipled. I think when we do the TODO to make the Log and
+// Blob storage traits async, this will clear up a bit.
+//
+// NB: This is pretty duplicative of start, but that's okay. It's already fairly
+// different and it's only going to get more so. DRYing this up wouldn't really
+// decrease the chance of bugs but it would decrease readability.
+pub fn start_read<L, B>(
+    blob: B,
+    build: BuildInfo,
+    reg: &MetricsRegistry,
+    pool: Option<Arc<Runtime>>,
```

Comment:
nit: I'm guessing `pool` comes from when this was a thread pool and we didn't yet use an async runtime? Might be good to change it to `runtime` or `async_runtime` now? Same for `start`, above.

Reply:
No need to change, that reasoning also makes sense to me.

Reply:
fixed in the other PR!